aesara/tensor/basic.py

```diff
@@ -260,7 +260,7 @@ def numpy_scalar(data):
         assert np.all(np.array([]) == data)
         raise EmptyConstantError()
     try:
-        np.complex(data)  # works for all numeric scalars
+        np.complex128(data)  # works for all numeric scalars
```

Comment:
What's this change about? Could it be the reason for failing tests?

Reply:
For this change, I kept on coming across the following numpy deprecation error when running `pytest tests/tensor/test_basic.py`:

<details>
<summary>
Error
</summary>

```python
========================================== warnings summary ==========================================
aesara/scalar/basic.py:2416
  /Users/CloudChaoszero/Documents/Projects-Dev/Theano-PyMC/aesara/scalar/basic.py:2416: DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    self.ctor = getattr(np, o_type.dtype)

tests/tensor/test_basic.py: 3247 warnings
  /Users/CloudChaoszero/Documents/Projects-Dev/Theano-PyMC/aesara/tensor/basic.py:263: DeprecationWarning: `np.complex` is a deprecated alias for the builtin `complex`. To silence this warning, use `complex` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.complex128` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    np.complex(data)  # works for all numeric scalars
```

</details>

The numpy Bool datatype is affected as well.

Using ``numpy 1.20.0``